### PR TITLE
Update schemas to match the Paris edition of the protocol

### LINKED ISSUE #130: Update to Paris edition of protocol
### Description
The [Paris edition of the Eiffel protocol](https://github.com/eiffel-community/eiffel/releases/tag/edition-paris) has been released (see https://github.com/eiffel-community/eiffel/pull/253) and REMReM should be updated to that edition. The CD and ArtP events have been bumped from 3.0.0 to 3.1.0.

### Motivation
The Paris edition includes a few interesting changes that we'd like to be able to use with REMReM.

### Exemplification
We have a need to publish artifact events for multi-file artifacts and support for that wasn't introduced until the Paris edition of the protocol.

### Benefits
REMReM clients get access to the latest event versions.

### Possible Drawbacks
None. The changes in Paris are backwards compatible.

### PR DESCRIPTION
**Pull request by magnusbaeck**
### Applicable Issues
Closes #130 

### Description of the Change
Updating schemas to the Paris edition effectively means stepping up ArtP and CD to v3.1.0, but only the former actually affects the schemas.

Some of the generated files that were present prior to this commit appear to have been modified after they were generated, i.e. regenerating them introduced various non-consequential changes (whitespace and/or line endings).

The copyright header in each file was retained though, although we should update the code that generates the files to keep the header. I've filed #133 for this.

### Alternate Designs
None.

### Benefits
Latest protocol edition available to REMReM users.

### Possible Drawbacks
None.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I    have the right to submit it under the open source license    indicated in the file; or

(b) The contribution is based upon previous work that, to the best    of my knowledge, is covered under an appropriate open source    license and I have the right under that license to submit that    work with modifications, whether created in whole or in part    by me, under the same open source license (unless I am    permitted to submit under a different license), as indicated    in the file; or

(c) The contribution was provided directly to me by some other    person who certified (a), (b) or (c) and I have not modified    it.

(d) I understand and agree that this project and the contribution    are public and that a record of the contribution (including all    personal information I submit with it, including my sign-off) is    maintained indefinitely and may be redistributed consistent with    this project or the open source license(s) involved.

Signed-off-by: Magnus Bäck \<magnus.back@axis.com\>

### CODE REVIEW

**Review by SantoshNC68 — COMMENTED** [review]
Is it a minor change or a major change ?
Updating the protocol version can be considered a major change?

**Comment by magnusbaeck** [comment]
If you're thinking of what component of the version number to bump I'd say it's an update of the minor version. I presume you're using semantic versioning, and a major version bump would imply a backwards-incompatible change. The update from Agen-1 to Paris is backwards-compatible.

**Comment by SantoshNC68** [comment]
> If you're thinking of what component of the version number to bump I'd say it's an update of the minor version. I presume you're using semantic versioning, and a major version bump would imply a backwards-incompatible change. The update from Agen-1 to Paris is backwards-compatible.

Ok, I was refering to semantic versioning. I see that the changes are backward compatible and since it was a new version wanted to check if we needed to release a major vesion.


Now I approve the changes.

**Review by SantoshNC68 — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -111,7 +111,7 @@
                             <mainClass>com.ericsson.eiffel.remrem.semantics.clone.PrepareLocalEiffelSchemas</mainClass>
                             <arguments>
```
Comment (SantoshNC68) [inline pom.xml:112]: You need to also update the version in this file

Comment (SantoshNC68) [reply pom.xml:112]: to 2.2.0

Comment (magnusbaeck) [reply pom.xml:112]: Right, thanks. Done.

**Review by SantoshNC68 — APPROVED** [review]

**Comment by magnusbaeck** [comment]
@eiffel-community/eiffel-remrem-maintainers, please have a look at this PR so that we can update REMReM to support the current edition of the protocol.

**Review by raja-maragani — CHANGES_REQUESTED** [review]
Most of changes are code format changes, Could you please revert the format change and place only changes related to Paris?

**Comment by magnusbaeck** [comment]
> Most of changes are code format changes, Could you please revert the format change and place only changes related to Paris?

No. These are autogenerated files and hand-editing them makes no sense at all. But I should've split this commit in two; one that generated the files from the old edition of the schemas (resulting in a pure reformatting commit) and one that stepped up to Paris. The end state would've been the same though.

Updating the code in this repository when a new edition is released should be something that can be automated entirely. Instead I spent hours dealing with the incorrect line endings (mix of CR/LF and LF), retaining copyright headers, and fixing the bugs that caused the generation to only work on Windows (fixed in #135).

**Comment by raja-maragani** [comment]
> > Most of changes are code format changes, Could you please revert the format change and place only changes related to Paris?
> 
> No. These are autogenerated files and hand-editing them makes no sense at all. But I should've split this commit in two; one that generated the files from the old edition of the schemas (resulting in a pure reformatting commit) and one that stepped up to Paris. The end state would've been the same though.
> 
> Updating the code in this repository when a new edition is released should be something that can be automated entirely. Instead I spent hours dealing with the incorrect line endings (mix of CR/LF and LF), retaining copyright headers, and fixing the bugs that caused the generation to only work on Windows (fixed in #135).

@magnusbaeck Great! Please make two PR's.

**Review by m-linner-ericsson — APPROVED** [review]

**Comment by m-linner-ericsson** [comment]
@raja-maragani Are you waiting for this to be split into two PRs?

**Comment by raja-maragani** [comment]
> @raja-maragani Are you waiting for this to be split into two PRs?

Yes @m-linner-ericsson

**Comment by magnusbaeck** [comment]
I've filed #136 for fixing the autogenerated files. I hope I'll be able to work on that tomorrow. Once that PR is in I'll update this one.

**Review by raja-maragani — APPROVED** [review]
+1

**Review by SantoshNC68 — APPROVED** [review]

**Pull request merged**